### PR TITLE
feat: implement hint backbone

### PR DESCRIPTION
**Pull request by jjeffcaii**
**What this PR does**: 
- implement hint backbone
- some bug fix
- optimizer refactor

**Which issue(s) this PR fixes**: 

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by chuntaojun — APPROVED** [review]

File: pkg/proto/hint/hint.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+// Hint represents a Hint, a valid Hint should include type and input kv pairs.
+//
+// Follow the format below:
+//   - without inputs: YOUR_HINT()
+//   - with non-keyed inputs: YOUR_HINT(foo,bar,quz)
+//   - with keyed inputs: YOUR_HINT(x=foo,y=bar,z=quz)
+//
+type Hint struct {
+	Typ    Type
```
Comment (dongzl) [inline pkg/proto/hint/hint.go:56]: why the variable name is `Typ`?

File: pkg/proto/hint/hint.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
+package hint
```
Comment (dongzl) [inline pkg/proto/hint/hint.go:1]: add license header.

**Review by dongzl — APPROVED** [review]

**Comment by dongzl** [comment]
Hi @jjeffcaii , I find `optimizer.go` to split to multi optimizer file.

```
optimizer_show_column.go
optimizer_select.go
...
```

I think the package is `optimize`, maybe we can remove the file prefix `optimize`. It's excess.

File: pkg/proto/schema_manager/loader_test.go [hunk 3]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package schema_manager
+package schema_manager_test
```
Comment (AlexStocks) [inline pkg/proto/schema_manager/loader_test.go:18]: package 名字怎么能有下划线？

File: pkg/proto/hint/hint.go (hunk 2 above)
Comment (jjeffcaii) [reply pkg/proto/hint/hint.go:18]: fixed

File: pkg/proto/schema_manager/loader_test.go (hunk 3 above)
Comment (jjeffcaii) [reply pkg/proto/schema_manager/loader_test.go:18]: 这个`_test`后缀为了消除循环imports，因为 `runtime` 依赖了 `schema_manager`, 而`schema_manager`虽然没有导入`runtime`，但是这个单测里依赖了，导致出现cycle imports， 所以需要显式加上`_test`后缀，消除这个单测文件的问题。

File: pkg/proto/hint/hint.go (hunk 1 above)
Comment (jjeffcaii) [reply pkg/proto/hint/hint.go:73]: Just a personal hobby, will be renamed to `Type`.

**Review by PangXing — APPROVED** [review]

**Review by Lvnszn — APPROVED** [review]

**Pull request merged**